Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -125,15 +125,30 @@ public struct HTTPRequestHead: Equatable {
 
 extension HTTPRequestHead: @unchecked Sendable {}
 
-/// The parts of a complete HTTP message, either request or response.
-///
-/// A HTTP message is made up of a request or status line with several headers,
-/// encoded by `.head`, zero or more body parts, and optionally some trailers. To
-/// indicate that a complete HTTP message has been sent or received, we use `.end`,
-/// which may also contain any trailers that make up the message.
+/// The parts of a complete HTTP message, representing either a request or a response.
+/// 
+/// An HTTP message is made up of:
+/// - a request or status line with several headers, encoded by a single ``HTTPPart/head(_:)`` part,
+/// - zero or more ``HTTPPart/body(_:)`` parts,
+/// - and some optional trailers (represented as headers) in a single ``HTTPPart/end(_:)`` part.
+/// 
+/// To indicate that a complete HTTP message has been sent or received,
+/// an ``HTTPPart/end(_:)`` part should always be used, even when no trailers are included.
```

Comment:
"should" is a suggestion, "must" is a requirement so let's use that here:

```suggestion
/// an ``HTTPPart/end(_:)`` part must be used, even when no trailers are included.
```

Reply:
Good callout, thanks!